connector/jobrunner/runner.py

```diff
@@ -194,6 +195,15 @@ def __init__(self, db_name):
         self.conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
         self.has_connector = self._has_connector()
         if self.has_connector:
+            if not self._lock():
+                _logger.exception(
+                    "Jobrunner lock was not granted. Is there another "
+                    "jobrunner active in database %s? Trying to stop Odoo.",
+                    self.db_name)
+                os.kill(openerp.service.server.server.pid, signal.SIGINT)
```

Comment:
Is it possible that this is not executed correctly? Should we not do a try catch around this and perhaps take additional actions to kill Odoo entirely?

Reply:
That's what the raise is for after this. If we cannot kill Odoo for some reason, the exception is raised, the jobrunner sleeps for a little while and tries again. If the main Odoo process does not respond to SIGINT repeatedly, then you probably have bigger problems.

Reply:
BTW the call itself, even if it does not guarantee that Odoo itself is stopped will not easily raise an exception so try/catch would be void.

Reply:
Ok, great! Thanks for the clarification.